pkg/gatewayserver/io/basicstationlns/messages/upstream.go

```diff
@@ -181,7 +168,6 @@ func (req *JoinRequest) ToUplinkMessage(ids ttnpb.GatewayIdentifiers, bandID str
 		RSSI:               req.RadioMetaData.UpInfo.RSSI,
 		ChannelRSSI:        req.RadioMetaData.UpInfo.RSSI,
 		SNR:                req.RadioMetaData.UpInfo.SNR,
```

Comment:
You'll need to add `RCtx` as the Antenna Index here then;
```
AntennaIndex:       uint32(req.RadioMetaData.UpInfo.RCtx)
```

Reply:
Why "then"? The `AntennaIndex` was never set, also not via the uplink token.

But I replaced setting and getting `RCtx` using `AntennaIndex` as backing. I think that's an important fix anyway for multi-antenna Basic Station gateways, as `AntennaIndex` was left to `0`, but it was set in the uplink token which would get overwritten by GS further in the chain, with `AntennaIndex` still at `0`, breaking downlink because the timestamp is slightly off.

Reply:
"then" because you removed the uplink token handling here 😉 .